### PR TITLE
Adding scalar function JsonPathArray to extract arrays from json

### PR DESCRIPTION
**Pull request by fx19880617**
## Description
Adding scalar function JsonPathArray to extract arrays from JSON.

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.jayway.jsonpath.JsonPath;
 import java.util.Map;
+import net.minidev.json.JSONArray;
```
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java:24]: it's the internal impl for jsonPath lib

File: pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java [hunk 2]
```diff
@@ -65,4 +65,42 @@ public void testJsonFunction()
...
+
+  @Test
+  public void testJsonFunctionOnJsonArray()
+      throws JsonProcessingException {
+    String jsonArrayString =
+        "[\n" +
+        "        {\n" +
+        "            \"name\": \"maths\",\n" +
+        "        \"grade\": \"A\"\n" +
+        "        },\n" +
```
Comment (siddharthteotia) [inline pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java:97]: Can we try nested array as well?
Something like subjects 
`{ name, array of lastTwoSemesterGrades}`

Comment (siddharthteotia) [reply pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java:97]: Also, scalar functions should also have e2e query execution unit tests. Alternatively, may be add a query compilation unit test as well to ensure it gets recognized/compiled as scalar

Comment (fx19880617) [reply pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java:103]: added.

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fx19880617) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java]: Update the jayway config to use jackson by default.

File: pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java (hunk 2 above)
Comment (siddharthteotia) [reply pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java:109]: Should have mentioned earlier itself. Can we please add one test for empty array as well?

Also,  Can we chain the jsonPathArray with jsonExtract as a transform and extract a particular value out of the array? Or for such cases, the recommended way is to directly use jsonExtract as opposed to doing jsonExtract(jsonPathArray(....)) ?

**Review by siddharthteotia — APPROVED** [review]
